arm/.global/global.module.tests.ps1

```diff
@@ -763,6 +763,19 @@ Describe 'Deployment template tests' -Tag Template {
                 $TenantIdKeyCount -eq $TenantIdValueCount | Should -Be $true -Because ("Parameter file should not contain the Tenant ID guid, instead should reference a token value '$ParameterFileTokenName'")
             }
         }
+
+        It '[<moduleFolderName>] [Tokens] Parameter files, resource name should contain the namePrefix token' -TestCases $deploymentFolderTestCases {
+            param (
+                [hashtable[]] $ParameterFileTestCases
+            )
+            foreach ($ParameterFileTestCase in $ParameterFileTestCases) {
+                $ParameterFileTokenName = -join ($ParameterFileTestCase.tokenSettings.tokenPrefix, 'namePrefix', $ParameterFileTestCase.tokenSettings.tokenSuffix)
+                $ParameterFileContent = Get-Content -Path $ParameterFileTestCase.parameterFile_Path
+                $namePrefixValue = $ParameterFileTestCase.tokenSettings.localTokens.tokens | Where-Object -Property name -EQ 'namePrefix' | Select-Object -ExpandProperty value
+                $namePrefixValueCount = ($ParameterFileContent | Select-String -Pattern "$namePrefixValue" -AllMatches).Matches.Count
+                $namePrefixValueCount -eq 0 | Should -Be $true -Because ("Parameter file should contain the token: $ParameterFileTokenName instead of value: $namePrefixValue")
```

Comment:
@ahmadabdalla , how big of a chance you see in this 'accidently' breaking deployments? Just thougt of a prefix like `kvlt` (dumb example, I know) - could that not potentially clash with just a resource type reference in a parameter name (e.g. if somebody uses `<namePrefix>-az-kvlt-min>`?

Reply:
as discussed. Good point, I will remove the pester test for now, and open up an issue for discovery as a backlog item. https://github.com/Azure/ResourceModules/issues/951